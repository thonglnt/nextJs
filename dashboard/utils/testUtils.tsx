import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
