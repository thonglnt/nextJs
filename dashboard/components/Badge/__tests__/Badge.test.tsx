import MailIcon from '@mui/icons-material/Mail'

// Utils
import { render } from '@utils/testUtils'

// Components
import BadgeVisibility, { IBadge } from '..'

describe('Badge Component', () => {
  const mockProps: IBadge = {
    children: <MailIcon color="action" />,
    content: 9,
    color: 'error',
    invisible: false,
  }

  it('Should show match Badge DOM Snapshot', () => {
    const { container } = render(<BadgeVisibility {...mockProps} />)

    expect(container).toMatchSnapshot()
  })
})
