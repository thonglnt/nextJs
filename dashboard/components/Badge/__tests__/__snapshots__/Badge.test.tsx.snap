// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component Should show match Badge DOM Snapshot 1`] = `
<div>
  <span
    class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeMedium css-11pbyhm-MuiSvgIcon-root"
      data-testid="MailIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
    <span
      class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightCircular MuiBadge-overlapCircular MuiBadge-colorError MuiBadge-badge css-10tn45-MuiBadge-badge"
    >
      9
    </span>
  </span>
</div>
`;
